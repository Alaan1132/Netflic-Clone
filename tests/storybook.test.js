import initStoryshots from '@storybook/addon-storyshots';

//  Created a global mock to avoid the error for the tests:
//  --- TypeError: Cannot read property 'STORYBOOK_HOOKS_CONTEXT' of undefined ---
//  https://github.com/storybookjs/storybook/issues/8083#issuecomment-533184737
jest.mock('global', () => Object.assign(global, { window: { STORYBOOK_HOOKS_CONTEXT: '' } }));

initStoryshots();
