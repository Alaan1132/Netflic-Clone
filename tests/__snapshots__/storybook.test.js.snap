// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Buttons Button With Icon 1`] = `
<button
  className="btn btnIcon primary size-md isRounded"
>
  <div
    className="icon"
  >
    <svg
      className="color-inverted"
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc>
        caretDown
      </desc>
      <path
        d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
      />
    </svg>
  </div>
</button>
`;

exports[`Storyshots Atoms/Buttons Button With Icon And Label 1`] = `
<button
  className="btn btnIcon primary size-md"
>
  Comenzar
  <div
    className="icon"
  >
    <svg
      className="color-inverted"
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc>
        caretDown
      </desc>
      <path
        d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
      />
    </svg>
  </div>
</button>
`;

exports[`Storyshots Atoms/Buttons Default 1`] = `
<button
  className="btn primary size-md"
>
  Siguiente
</button>
`;

exports[`Storyshots Atoms/Buttons Types 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-end",
      "display": "flex",
      "gap": "2rem",
    }
  }
>
  <button
    className="btn primary size-md"
  >
    Siguiente
  </button>
  <button
    className="btn secondary size-md"
  >
    Siguiente
  </button>
  <button
    className="btn tertiary size-md"
  >
    Siguiente
  </button>
  <button
    className="btn outline-primary size-md"
  >
    Siguiente
  </button>
  <button
    className="btn outline-muted size-md"
  >
    Siguiente
  </button>
</div>
`;

exports[`Storyshots Atoms/Checkbox Default 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="checkbox"
  >
    <div
      className="hidden"
    >
      <svg
        className="color-base"
        style={
          Object {
            "height": 22,
            "width": 22,
          }
        }
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          check
        </desc>
        <path
          d="M3.707 12.293l-1.414 1.414L8 19.414 21.707 5.707l-1.414-1.414L8 16.586z"
        />
      </svg>
    </div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <p
    className="paragraph size-md color-base weight-normal"
  >
    Acepto
  </p>
</div>
`;

exports[`Storyshots Atoms/Heading Default 1`] = `
<h1
  className="heading color-base weight-bold"
>
  Hello
</h1>
`;

exports[`Storyshots Atoms/Icon Default 1`] = `
<div
  className="icon"
>
  <svg
    className="color-base"
    style={
      Object {
        "height": 22,
        "width": 22,
      }
    }
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <desc>
      caretDown
    </desc>
    <path
      d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
    />
  </svg>
</div>
`;

exports[`Storyshots Atoms/Input Dark 1`] = `
<div
  className="group"
>
  <div
    className="input__group"
  >
    <input
      autoComplete="off"
      className="input input__dark"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      type="password"
      value=""
    />
    <label
      className="label label__dark"
      htmlFor="password"
    >
      Insert your password
    </label>
  </div>
  <p
    className="paragraph size-md weight-normal"
  >
    Error
  </p>
</div>
`;

exports[`Storyshots Atoms/Input Email 1`] = `
<div
  className="group"
>
  <div
    className="input__group"
  >
    <input
      autoComplete="off"
      className="input"
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      type="email"
      value=""
    />
    <label
      className="label"
      htmlFor="email"
    >
      Insert your email
    </label>
  </div>
  <p
    className="paragraph size-md weight-normal"
  >
    Error
  </p>
</div>
`;

exports[`Storyshots Atoms/Input Email Or Number 1`] = `
<div
  className="group"
>
  <div
    className="input__group"
  >
    <input
      autoComplete="off"
      className="input"
      id="emailOrPhone"
      name="emailOrPhone"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      type="text"
      value=""
    />
    <label
      className="label"
      htmlFor="emailOrPhone"
    >
      Insert your email or number
    </label>
  </div>
  <p
    className="paragraph size-md weight-normal"
  >
    Error
  </p>
</div>
`;

exports[`Storyshots Atoms/Input Password 1`] = `
<div
  className="group"
>
  <div
    className="input__group"
  >
    <input
      autoComplete="off"
      className="input"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyUp={[Function]}
      type="password"
      value=""
    />
    <label
      className="label"
      htmlFor="password"
    >
      Insert your password
    </label>
  </div>
  <p
    className="paragraph size-md weight-normal"
  >
    Error
  </p>
</div>
`;

exports[`Storyshots Atoms/Links Default 1`] = `
<a
  className="link size-md"
>
  Link
</a>
`;

exports[`Storyshots Atoms/Links Sign Up 1`] = `
<a
  className="link primary size-md"
>
  Iniciar sesion
</a>
`;

exports[`Storyshots Atoms/Paragraph Default 1`] = `
<p
  className="paragraph size-md color-base weight-normal"
>
  Hello World
</p>
`;

exports[`Storyshots Atoms/Paragraph Sizes 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-end",
      "display": "flex",
      "gap": "2rem",
    }
  }
>
  <p
    className="paragraph size-sm color-base weight-normal"
  >
    Hello World
  </p>
  <p
    className="paragraph size-md color-base weight-normal"
  >
    Hello World
  </p>
  <p
    className="paragraph size-lg color-base weight-normal"
  >
    Hello World
  </p>
</div>
`;

exports[`Storyshots Atoms/Picture Default 1`] = `
<picture
  className="picture"
>
  <img
    alt="img"
    src="https://picsum.photos/id/1033/400/400"
    style={
      Object {
        "height": "auto",
        "maxWidth": 200,
      }
    }
  />
</picture>
`;

exports[`Storyshots Layout/Navbar Nav Landing Page 1`] = `
<nav
  className="navbar"
>
  <div
    className="inner_container"
  >
    <div
      className="primary"
    >
      <div
        className="logo"
      >
        <picture
          className="picture"
        >
          <img
            alt="img"
            src="https://res.cloudinary.com/desxpqy6n/image/upload/v1630518956/Netflix%20Clone/netflix-logo_kuii2i.png"
            style={
              Object {
                "height": "auto",
                "maxWidth": "100%",
              }
            }
          />
        </picture>
      </div>
    </div>
    <div
      className="secondary"
    >
      <div
        className="container"
      >
        <div
          className="icon"
          style={
            Object {
              "left": 5,
              "position": "absolute",
            }
          }
        >
          <svg
            className="color-inverted"
            style={
              Object {
                "height": 14,
                "width": 14,
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc>
              global
            </desc>
            <path
              d="M256 48C141.13 48 48 141.13 48 256s93.13 208 208 208 208-93.13 208-208S370.87 48 256 48z"
              stroke="currentColor"
              strokeMiterlimit="10"
              strokeWidth="26"
            />
            <path
              d="M256 48c-58.07 0-112.67 93.13-112.67 208S197.93 464 256 464s112.67-93.13 112.67-208S314.07 48 256 48z"
              stroke="currentColor"
              strokeMiterlimit="10"
              strokeWidth="26"
            />
            <path
              d="M117.33 117.33c38.24 27.15 86.38 43.34 138.67 43.34s100.43-16.19 138.67-43.34M394.67 394.67c-38.24-27.15-86.38-43.34-138.67-43.34s-100.43 16.19-138.67 43.34"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="26"
            />
            <path
              d="M256 48v416M464 256H48"
              stroke="currentColor"
              strokeMiterlimit="10"
              strokeWidth="26"
            />
          </svg>
        </div>
        <select
          className="select-language size-xs"
        >
          <option
            data-country="MX"
            data-language="es"
            lang="es"
            value="/mx/"
          >
            Espa√±ol
          </option>
          <option
            data-country="MX"
            data-language="en"
            lang="en"
            value="/mx-en/"
          >
            English
          </option>
        </select>
        <div
          className="icon"
          style={
            Object {
              "position": "absolute",
              "right": 5,
            }
          }
        >
          <svg
            className="color-inverted"
            style={
              Object {
                "height": 14,
                "width": 14,
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc>
              caretDown
            </desc>
            <path
              d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
            />
          </svg>
        </div>
      </div>
      <a
        className="link primary size-xs"
      >
        Iniciar sesion
      </a>
    </div>
  </div>
</nav>
`;

exports[`Storyshots Layout/Navbar Signup Navbar 1`] = `
<nav
  className="border--bottom"
>
  <div
    className="inner_container"
  >
    <div
      className="primary"
    >
      <div
        className="logo"
      >
        <picture
          className="picture"
        >
          <img
            alt="img"
            src="https://res.cloudinary.com/desxpqy6n/image/upload/v1630518956/Netflix%20Clone/netflix-logo_kuii2i.png"
            style={
              Object {
                "height": "auto",
                "maxWidth": "100%",
              }
            }
          />
        </picture>
      </div>
    </div>
    <div
      className="secondary"
    >
      <a
        className="link primary size-xs"
      >
        Cerrar sesion
      </a>
    </div>
  </div>
</nav>
`;

exports[`Storyshots Molecules/Dropdown Select Language 1`] = `
<div
  className="container"
>
  <div
    className="icon"
    style={
      Object {
        "left": 5,
        "position": "absolute",
      }
    }
  >
    <svg
      className="color-inverted"
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc>
        global
      </desc>
      <path
        d="M256 48C141.13 48 48 141.13 48 256s93.13 208 208 208 208-93.13 208-208S370.87 48 256 48z"
        stroke="currentColor"
        strokeMiterlimit="10"
        strokeWidth="26"
      />
      <path
        d="M256 48c-58.07 0-112.67 93.13-112.67 208S197.93 464 256 464s112.67-93.13 112.67-208S314.07 48 256 48z"
        stroke="currentColor"
        strokeMiterlimit="10"
        strokeWidth="26"
      />
      <path
        d="M117.33 117.33c38.24 27.15 86.38 43.34 138.67 43.34s100.43-16.19 138.67-43.34M394.67 394.67c-38.24-27.15-86.38-43.34-138.67-43.34s-100.43 16.19-138.67 43.34"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="26"
      />
      <path
        d="M256 48v416M464 256H48"
        stroke="currentColor"
        strokeMiterlimit="10"
        strokeWidth="26"
      />
    </svg>
  </div>
  <select
    className="select-language size-md"
  >
    <option
      data-country="MX"
      data-language="es"
      lang="es"
      value="/mx/"
    >
      Espa√±ol
    </option>
    <option
      data-country="MX"
      data-language="en"
      lang="en"
      value="/mx-en/"
    >
      English
    </option>
  </select>
  <div
    className="icon"
    style={
      Object {
        "position": "absolute",
        "right": 5,
      }
    }
  >
    <svg
      className="color-inverted"
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc>
        caretDown
      </desc>
      <path
        d="M98 190.06l139.78 163.12a24 24 0 0036.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
      />
    </svg>
  </div>
</div>
`;
